geode-core/src/main/java/org/apache/geode/internal/util/JavaWorkarounds.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.util;
+
+import java.util.Map;
+import java.util.function.Function;
+
+public class JavaWorkarounds {
+
+  // This is a workaround for computeIfAbsent not being concurrent in pre Java 9
+  // https://bugs.openjdk.java.net/browse/JDK-8161372
+  public static <K, V> V computeIfAbsent(Map<K, V> map, K key,
```

Comment:
Do you have a JMH benchmark, or similar, that shows this is really an improvement? I think it is obviously an improvement if the entry is expected to be there for the vast majority of the calls. This will have a more negative impact though if the use case expects the entry to not be there. In that use case we go through the get path twice.

Reply:
Can we classify our use cases of this new wrapper and ensure that they are of the case where the expectation is that there is a value more often than not?

Reply:
This was discovered in an adhoc test where we were unable to scale client queries.  The ticket has the stack from the thread dumps which helped identify the issue.
The cost to compute the first time will increase slightly as you stated, but if it's not called very often to get the benefit of a cache, is the performance degrade enough to 'matter'?  

I can limit this diff to only affect the two calls that I can see hampering our most common use cases, I just thought making them all the same would make it more consistent across the code base? 

A JMH benchmark can be written to hammer these methods directly, but that will just prove that the issues with computeIfAbsent  (in Pre Java 9) are blocking/slower when the value is expected to be present.

Reply:
If we are reasonably confident that all theses cases are read heavy then it’s all good.

Reply:
I'll still add the JMH tests since it's probably good to test some of these other areas out.  I'll follow up on the other calls to try to reason if the change has a negative impact.  I just realized my last comment probably felt like I was trying to cut the convo off.. whoops.